GCRCatalogs/dc2_metacal.py

```diff
@@ -33,12 +33,32 @@ class DC2MetacalCatalog(DC2DMCatalog):
     SCHEMA_FILENAME = 'schema.yaml'
     META_PATH = os.path.join(FILE_DIR, 'catalog_configs/_dc2_metacal_meta.yaml')
 
-    @staticmethod
-    def _generate_modifiers(dm_schema_version=3, bands='riz'):
+
+    def _subclass_init(self, **kwargs):
+        """
+        Overrides default init method to apply various corrections to the catalog
+        """
+        # Default values of reader parameters
+        self._flux_scaling = 1.0
+        self._bands = 'riz'
+
+        if kwargs.get('bands'):
+            self._bands = kwargs['bands']
+
+        if kwargs.get('fix_metacal_test3'):
+            # In Run 1.2i metacal_test3, the fluxes returned by metacal are not
+            # properly scaled by the pixel size, it's necessary to apply a
+            # 0.2**2 correction factor
+            self._flux_scaling = 1. / 0.2**2
```

Comment:
Lines 41-52. This is mostly just preference... Sorry for being annoying.  
```python
        self._bands = kwargs.get('bands', 'riz')
 
		self._flux_scaling = 1.0
		if kwargs.get('fix_metacal_test3'):
            # In Run 1.2i metacal_test3, the fluxes returned by metacal are not
            # properly scaled by the pixel size, it's necessary to apply a
            # 0.2**2 correction factor
            self._flux_scaling /= (0.2**2)
```

Reply:
lol, sure :-D